modules/ensemble/lib/action/upload_files_action.dart

```diff
@@ -145,6 +145,7 @@ Future<void> uploadFiles({
           : (error) =>
               ScreenController().executeAction(context, action.onError!),
       progressCallback: (progress) {
+        if (action.id == null) return;
```

Comment:
FileUploadAction.id seems to be required but is defined as optional in FileUploadAction class. Why not just make it required so you don't have to make these checks?

Reply:
Id is not required. It is used when user want to bind something. Like here if user want to listen to progress of uploading.

I have put check already, in rest of file upload. I saw this was missing throwing exceptions in logs saw just fixed it.